ddtrace/tracer/servicehash.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package tracer
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"sync"
+)
+
+const maxCacheSize = 1000
+
+var (
+	cache = make(map[string]string)
+	lock  sync.RWMutex
+)
+
+// Hash computes hash for `service`. Will use a local cache to avoid recomputation.
+func Hash(service string) string {
+	hash := getHashFromCache(service)
+	if len(hash) > 0 {
+		return hash
+	}
```

Comment:
```suggestion
	if h, ok := hashFromCache(service); ok {
		return h
	}
```
I don't know if there's any point to go through the trouble of re-using this `hash` variable.